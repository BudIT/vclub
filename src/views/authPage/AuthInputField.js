import React, { PropTypes } from 'react';
import styles from './AuthPage.css';

function AuthInputField(props) {
  const {
    input, type, placeholder, className, meta,
  } = props;

  const {
    touched, error,
  } = meta;

  return (
    <div>
      <div>
        <input
          {...input}
          type={type}
          placeholder={placeholder}
          className={className}
          autoFocus
        />
        {(touched && error)
        && <span className={styles.errors}>{error}</span>}
      </div>
    </div>
  );
}

AuthInputField.propTypes = {
  input: PropTypes.object.isRequired,
  type: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
  meta: PropTypes.shape({
    touched: PropTypes.bool.isRequired,
    error: PropTypes.string,
  }),
};

export default AuthInputField;
