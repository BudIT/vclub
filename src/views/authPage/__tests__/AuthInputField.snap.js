/* eslint import/no-extraneous-dependencies: ["error", {"devDependencies": true}] */
/* eslint-env jest */

import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';
import AuthInputField from '../AuthInputField';

const createProps = (touched, error) => ({
  input: {},
  type: 'text',
  placeholder: 'placeholder',
  className: 'name',
  meta: {
    touched,
    error,
  },
});

const shallowRenderer = ReactTestUtils.createRenderer();

test('<AuthInputField /> renders correctly when touched is false', () => {
  const props = createProps(false, 'some error');
  shallowRenderer.render(
    <AuthInputField
      {...props}
    />
  );
  expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
});

test('<AuthInputField /> renders correctly when touched is true', () => {
  const props = createProps(true, undefined);
  shallowRenderer.render(
    <AuthInputField
      {...props}
    />
  );
  expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
});

test('<AuthInputField /> renders correctly #3', () => {
  const props = createProps(true, 'some error');
  shallowRenderer.render(
    <AuthInputField
      {...props}
    />
  );
  expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
});
