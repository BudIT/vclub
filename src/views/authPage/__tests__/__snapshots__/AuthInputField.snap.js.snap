exports[`test <AuthInputField /> renders correctly #3 1`] = `
<div>
  <div>
    <input
      autoFocus={true}
      className="name"
      placeholder="placeholder"
      type="text" />
    <span
      className="errors">
      some error
    </span>
  </div>
</div>
`;

exports[`test <AuthInputField /> renders correctly when touched is false 1`] = `
<div>
  <div>
    <input
      autoFocus={true}
      className="name"
      placeholder="placeholder"
      type="text" />
  </div>
</div>
`;

exports[`test <AuthInputField /> renders correctly when touched is true 1`] = `
<div>
  <div>
    <input
      autoFocus={true}
      className="name"
      placeholder="placeholder"
      type="text" />
  </div>
</div>
`;
